src/workerd/server/server.c++

```diff
@@ -2023,6 +2023,14 @@ class Server::WorkerService final: public Service,
     }
 
     kj::Own<IoChannelFactory::ActorChannel> getActorChannel(Worker::Actor::Id id) {
+      KJ_IF_SOME(doId, id.tryGet<kj::Own<ActorIdFactory::ActorId>>()) {
+        // To emulate production, we have to recreate this ID.
+        ActorIdFactoryImpl::ActorIdImpl* idImpl =
+            dynamic_cast<ActorIdFactoryImpl::ActorIdImpl*>(doId.get());
+        KJ_ASSERT(idImpl != nullptr, "Unexpected ActorId type?");
```

Comment:
```suggestion
        KJ_ASSERT_NONNULL(idImpl, "Unexpected ActorId type?");
```

Reply:
KJ_ASSERT_NONNULL is meant to be used with kj::Maybe values. The comparison here is on a pointer, so using this macro is inappropriate.

We should arguably rename KJ_ASSERT_NONNULL to KJ_ASSERT_SOME.

Reply:
> We should arguably rename KJ_ASSERT_NONNULL to KJ_ASSERT_SOME.

TIL. Yes I agree.

> During coder reviews, it's extremely tempting to comment on every little thing that is not exactly the way the reviewer would have written it, but often it's not worth the time. If the code has a bug, is confusing, or contains maintainability pitfalls, then comment. 

I agree in principal but most of the time, such comments are educational and beneficial for reviewers more than the author itself.